# Pasando el problema a ejecutable con python:

In [2]:
import pandas as pd
import json


In [16]:
df = pd.read_csv('inundaciones.csv')
df.drop(columns=[ 'Comments','Description of Cause', 'Source', 'Evacuated'], inplace=True)
print(df["Departamento"].nunique())
df.head()

410


Serial  Date (YMD)  Event     Provincia  Code Provincia  \
0  2014-00077:003   29/6/2014  FLOOD      Misiones              54   
1  2014-00033:003    6/2/2014  FLOOD  Buenos Aires               6   
2  2014-00107:002  28/10/2014  FLOOD  Buenos Aires               6   
3      2014-00107  28/10/2014  FLOOD  Buenos Aires               6   
4      2014-00040   12/2/2014  FLOOD       Mendoza              50   

  Departamento  Code Departamento       Cause  DataCards        lat  \
0      Capital            54028.0  Llu y desb          1 -27.476595   
1      Berisso             6098.0   Llu y vie          1 -34.845876   
2   La Matanza             6427.0  Llu y desb          1 -34.714173   
3          NaN                NaN  Llu y desb          1        NaN   
4      Capital            50007.0  Llu y desb          1 -32.877494   

         lon                     rio1.nombre  rio1.distancia  \
0 -55.921700                         Posadas    8.071972e+00   
1 -57.879733                        La Plata    7.086465e-01   
2 -58.508904  Aº Medrano Reser. - Tecnópolis    8.891533e+00   
3        NaN                   nombre random    1.000000e+08   
4 -68.852787                 Mendoza - Guido    4.283537e+01   

                  rio2.nombre  rio2.distancia  
0                   Santa Ana       38.102938  
1                    La Plata        1.454843  
2                      Olivos       12.605877  
3                         NaN             NaN  
4  De los Patos - La Plateada       98.426041

### La cantidad de DEPARTAMENTOS únicos, es de 410. Este es el numero de zonas que va a poder colorear SIRENA. Pensando siempre para abajo, ya que seguro estos tienen datos faltantes.

In [20]:
#Haciendo columna que indique la influencia del rio de acuerdo a la distancia
df["influencia_rio"] = 1 / (1 + df["rio1.distancia"])
df.drop(columns=["rio2.nombre","rio2.distancia"], inplace=True)
df

Serial  Date (YMD)  Event        Provincia  Code Provincia  \
0     2014-00077:003   29/6/2014  FLOOD         Misiones              54   
1     2014-00033:003    6/2/2014  FLOOD     Buenos Aires               6   
2     2014-00107:002  28/10/2014  FLOOD     Buenos Aires               6   
3         2014-00107  28/10/2014  FLOOD     Buenos Aires               6   
4         2014-00040   12/2/2014  FLOOD          Mendoza              50   
...              ...         ...    ...              ...             ...   
6992          4558-b    4/1/1970  FLOOD            Salta              66   
6993          4550-b    2/1/1970  FLOOD          Mendoza              50   
6994          4550-a    2/1/1970  FLOOD          Mendoza              50   
6995          4633-a    2/1/1970  FLOOD  Capital Federal               2   
6996          4551-a    2/1/1970  FLOOD     Buenos Aires               6   

      Departamento  Code Departamento       Cause  DataCards        lat  \
0          Capital            54028.0  Llu y desb          1 -27.476595   
1          Berisso             6098.0   Llu y vie          1 -34.845876   
2       La Matanza             6427.0  Llu y desb          1 -34.714173   
3              NaN                NaN  Llu y desb          1        NaN   
4          Capital            50007.0  Llu y desb          1 -32.877494   
...            ...                ...         ...        ...        ...   
6992      Chicoana            66042.0        RAIN          1 -25.153392   
6993     Las Heras            50049.0        RAIN          1 -32.240089   
6994       Capital            50007.0        RAIN          1 -32.877494   
6995      Comuna 1             2001.0        RAIN          1 -34.625048   
6996  Bahia Blanca             6056.0        RAIN          1 -38.683875   

            lon                     rio1.nombre  rio1.distancia  \
0    -55.921700                         Posadas    8.071972e+00   
1    -57.879733                        La Plata    7.086465e-01   
2    -58.508904  Aº Medrano Reser. - Tecnópolis    8.891533e+00   
3           NaN                   nombre random    1.000000e+08   
4    -68.852787                 Mendoza - Guido    4.283537e+01   
...         ...                             ...             ...   
6992 -65.693632             Arenales - El Encon    1.811095e+01   
6993 -69.455315      De los Patos - La Plateada    2.727939e+01   
6994 -68.852787                 Mendoza - Guido    4.283537e+01   
6995 -58.384387                    Buenos Aires    4.079546e+00   
6996 -62.215956           Sauce Grande - RN Nº3    5.785154e+01   

      influencia_rio  
0       1.102296e-01  
1       5.852586e-01  
2       1.010966e-01  
3       1.000000e-08  
4       2.281263e-02  
...              ...  
6992    5.232602e-02  
6993    3.536145e-02  
6994    2.281263e-02  
6995    1.968680e-01  
6996    1.699191e-02  

[6997 rows x 14 columns]